### ISSUE TITLE
Can't install apigen v2.8.x

### ISSUE THREAD

**Issue by seeafish** [rackspace/php-opencloud#581]
Wanted to work on this repo, but haven't been able to get all the dev composer packages installed. It always fails on apigen/apigen. Looks like this is an issue, although no solution other than upgrading was provided upstream:
https://github.com/apigen/apigen/issues/466

I don't get exactly the same error (see below), but have asked around, and it seems like that version can't be reliably installed anymore. I should also note that v4.0.0 installs fine, although it upgrades a bunch of other packages.

My error output:
```
$ php composer.phar install
Loading composer repositories with package information
Installing dependencies (including require-dev)
  - Installing apigen/apigen (v2.8.1)
    Downloading: 100%
    Failed to download apigen/apigen from dist: Could not delete /vendor/apigen/apigen/6810c67042c9c00668422a9a0
e5b9a40:
    Now trying to download from source
  - Installing apigen/apigen (v2.8.1)
    Cloning bca0954e92621a48c6870be57115026a8d032706

  [RuntimeException]
  Failed to execute git checkout 'bca0954e92621a48c6870be57115026a8d032706' && git reset --hard 'bca0954e92621a48c687
  0be57115026a8d032706'
  fatal: Unable to create '/vendor/apigen/apigen/.git/index.lock': File exists.
  If no other git process is currently running, this probably means a
  git process crashed in this repository earlier. Make sure no other git
  process is running and remove the file manually to continue.
```
I have tried `--prefer-source` too and it simply reverses the order of issues.

Runing php 5.5.9 on Ubuntu 14.04.

**Comment by seeafish** [comment]
By the way, I've found this: https://github.com/composer/composer/issues/3843 by @ycombinator 
I can see that's been open for 20 days so far.
Can't we simply upgrade apigen to v4, since it was set to 2.8 seemingly to work with an EOL'd version of PHP: https://github.com/rackspace/php-opencloud/commit/9593334de68b44c5888b8ded47e4f5f497fca399

**Comment by jamiehannaford** [comment]
@seeafish I think the reason we didn't upgrade to v4 is because the minimum PHP version for that dep is 5.4, but our SDK has 5.3 as a minimum - changing that would be a breaking change. *Maybe* we could add it as a dev requirement, though:

```json
"require-dev": {
    "php" : "5.4"
}
```

I'm checking on vagrant now to see whether this will impact 5.3 users.

**Comment by seeafish** [comment]
Fixed via #582
